### PR TITLE
Smartgun top cover

### PR DESCRIPTION
**Pull request by stanalbatross**
## About The Pull Request

Ever wonder why our belt fed smartguns can be reloaded in a single click? Ever thought that seemed a bit silly especially when there's a scene in the movie with Vasquez reloading her smartgun where she clearly slams down the top cover? I thought so too, so here is this PR.

## Why It's Good For The Game

VIBRANT SOVL + movielike as FUCK + stops SGs from being a super unending stream of bullets by making the reload a bit more involved (as it should be for such a big gun)

## Changelog

:cl: Stan_Albatross
add: added a functioning top cover to the smartgun, can be opened and closed with alt-click. You must open the cover to remove the drum or replace it. You cannot fire the gun with the cover open. Thanks to Jooks for the sounds of opening/closing the cover.
fix: fixed being able to telepathically toggle the mode of laser designators.
/:cl:

### CODE REVIEW

**Comment by PhantomEpicness** [comment]
BASED FUCKING SOUL PR

**_* Disclaimer *_**
CM-SS13 is not related to the Disney-owned "Aliens" Franchise and IP This is an entirely fan-made project. In case of an investigation by any federal entity or similar, I do not have any involvement with this group nor with the people in it, I do not know how I am here, probably added by a thrid party, I do not support any actions by the member of this group. Any and all content made by this account were made without my permission and I have no knowledge of the usage of this GitHub™️ account to contribute to CM-SS13.

**Comment by ilikesaying** [comment]
Amazing pr

CM-SS13 is not related to the Disney-owned "Aliens" Franchise and IP. This is an entirely fan-made project. In case of an investigation by any federal entity or similar, I do not have any involvement with this group nor with the people in it, I do not know how I am here, probably added by a th**ir**d party, I do not support any actions by the member of this group. Any and all content made by this account were made without my permission and I have no knowledge of the usage of this GitHub ©️ account to contribute to CM-SS13.

**Comment by MusselMain** [comment]
Cool idea, but please don't make it like the M60, where you have to stand perfectly still and wait 3 seconds before you finish loading the new box, that'd be super fucked and turn this cool RP merge into a SG nerf

**Comment by IngloriouslyBad** [comment]
> Cool idea, but please don't make it like the M60, where you have to stand perfectly still and wait 3 seconds before you finish loading the new box, that'd be super fucked and turn this cool RP merge into a SG nerf

This is a nerf though. They're just using the new meme "soul" to push it through... and "RP" and "movie-like" as a keynote for doing it as well

**Comment by Shad0vvs** [comment]
Bruh

**Comment by Geevies** [comment]
> > Cool idea, but please don't make it like the M60, where you have to stand perfectly still and wait 3 seconds before you finish loading the new box, that'd be super fucked and turn this cool RP merge into a SG nerf
> 
> This is a nerf though. They're just using the new meme "soul" to push it through... and "RP" and "movie-like" as a keynote for doing it as well

they're onto us stan we need to fall back to plan B

this PR improves my immersion

**Comment by hotimep** [comment]
will there be a windup to opening the cover/closing it/putting a mag in or will it be all instant

**Comment by stanalbatross** [comment]
> will there be a windup to opening the cover/closing it/putting a mag in or will it be all instant

read the code (it is instant)

if people spam the sound I will not hesitate to add a cooldown between spam opening/closing

**Review by Shad0vvs — APPROVED** [review]

File: code/modules/projectiles/guns/specialist.dm
Comment (Watermelon914) [inline code/modules/projectiles/guns/specialist.dm:416]: `return ..()` and do this for the rest of them

**Comment by Watermelon914** [comment]
This doesn't really qualify as a balance PR because it's not made with the intention to change any playstyles and it doesn't directly change any values, but to rework the reload mechanic to have more depth (as it's no longer a click and wait minigame, it's now similar to reloading a regular gun)

Otherwise you can categorise any feature PR that remotely makes any change to anything related to combat as a balance change. Instead, it's better to categorise a balance change as something that's made with the intention to shake up the balance in some way, rather than putting it on feature PRs that you want prior approval for.
If you have an issue with the PR on a balance level, you're still free to make that complaint and slap a DNM tag on it so that both sides can properly discuss any issues that need to be resolved or compromised on without it being shadow-merged by a maintainer.

**Pull request merged**
